gdnative-core/src/export/method.rs

```diff
@@ -284,16 +287,303 @@ impl<'a> Varargs<'a> {
         let args = std::mem::transmute::<&[*mut sys::godot_variant], &[&Variant]>(args);
         Self {
             idx: 0,
-            iter: args.iter(),
+            args,
+            offset_index: 0,
+        }
+    }
+
+    /// Check the length of arguments.
+    /// See `get()`, `get_opt()` or `get_rest()` for examples.
+    ///
+    /// # Errors
+    /// Returns an error if the length of arguments is outside the specified range.
+    #[inline]
+    pub fn check_length(&self, bounds: impl RangeBounds<usize>) -> Result<(), ArgumentLengthError> {
+        let passed = self.args.len();
+        if bounds.contains(&passed) {
+            Ok(())
+        } else {
+            Err(ArgumentLengthError::new(passed, bounds))
+        }
+    }
+
+    /// Returns the type-converted value at the specified argument position.
+    ///
+    /// # Errors
+    /// Returns an error if the conversion fails or the argument is not set.
+    ///
+    /// # Examples
+    /// ```
+    /// # fn foo(args: gdnative::export::Varargs) -> Result<(), Box<dyn std::error::Error>> {
+    ///     args.check_length(2..=2)?;
+    ///     let a: usize = args.get(0)?;
+    ///     let b: i64 = args.get(1)?;
+    /// #   Ok(())
+    /// # }
+    /// ```
+    #[inline]
+    pub fn get<T: FromVariant>(&self, index: usize) -> Result<T, ArgumentTypeError> {
+        let relative_index = index;
+        let actual_index = index + self.offset_index;
+
+        match self.args.get(relative_index) {
+            Some(v) => match T::from_variant(v) {
+                Ok(ok) => Ok(ok),
+                Err(err) => Err(ArgumentTypeError::new(actual_index, err)),
+            },
+            None => {
+                let err = FromVariantError::Custom("Argument is not set".to_owned());
+                Err(ArgumentTypeError::new(actual_index, err))
+            }
+        }
+    }
+
+    /// Returns the type-converted value at the specified argument position.
+    /// Returns `None` if the argument is not set.
```

Comment:
Maybe mention that this is for optional parameters?

Also, we should probably clearly distinguish parameters (at declaration site) and arguments (at call site).
```rs
    /// Returns the type-converted value at the specified (optional) parameter position.
    /// This is for optional parameters; the method returns `Ok(None)` if the argument is not set.
```

For `get` on the other hand, you could mention:
```rs
    /// Returns the type-converted value at the specified parameter position.
    /// This is for required parameters; the method returns `Err` if the argument is missing.
```